### ISSUE TITLE
Handle filenames with multiple periods in

### ISSUE THREAD

**Issue by ns-rse** [AFM-SPM/TopoStats#271]
**Describe the bug**

When a filename has two periods in it we get crashes e.g. `mini.circle.spm`...

```
Traceback (most recent call last):
  File "/usr/lib/python3.10/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/home/neil/work/projects/topostats/TopoStats/topostats/run_topostats.py", line 305, in process_scan
    plot_and_save(
  File "/home/neil/work/projects/topostats/TopoStats/topostats/plottingfuncs.py", line 92, in plot_and_save
    plt.savefig(output_dir / filename)
  File "/home/neil/.virtualenvs/topostats/lib/python3.10/site-packages/matplotlib/pyplot.py", line 979, in savefig
    res = fig.savefig(*args, **kwargs)
  File "/home/neil/.virtualenvs/topostats/lib/python3.10/site-packages/matplotlib/figure.py", line 3046, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "/home/neil/.virtualenvs/topostats/lib/python3.10/site-packages/matplotlib/backend_bases.py", line 2259, in print_figure
    canvas = self._get_output_canvas(backend, format)
  File "/home/neil/.virtualenvs/topostats/lib/python3.10/site-packages/matplotlib/backend_bases.py", line 2188, in _get_output_canvas
    raise ValueError(
ValueError: Format 'circle_processed' is not supported (supported formats: eps, jpeg, jpg, pdf, pgf, png, ps, raw, rgba, svg, svgz, tif, tiff)
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/neil/.virtualenvs/topostats/bin/run_topostats", line 33, in <module>
    sys.exit(load_entry_point('topostats', 'console_scripts', 'run_topostats')())
  File "/home/neil/work/projects/topostats/TopoStats/topostats/run_topostats.py", line 394, in main
    for img, result in pool.imap_unordered(processing_function, img_files):
  File "/usr/lib/python3.10/multiprocessing/pool.py", line 873, in next
    raise value
ValueError: Format 'circle_processed' is not supported (supported formats: eps, jpeg, jpg, pdf, pgf, png, ps, raw, rgba, svg, svgz, tif, tiff)
```

This should be handled automatically.

A simple solution is in parsing/extracting the `filename` in the `Filters` class and to substitute the `.` for `_`.

**Comment by MaxGamill-Sheffield** [comment]
Should be resolved in #280

**Comment by MaxGamill-Sheffield** [comment]
Closed by #263 and a hotfix #286
